### PR TITLE
Add changelog entry for crate object[] -> pg json[] remapping.

### PR DESCRIPTION
**Pull request by kovrus**
## Summary of the changes / Why this improves CrateDB

See https://github.com/crate/crate/pull/9367

References https://github.com/crate/crate-npgsql/pull/37

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [ ] Touched code is covered by tests
 - [ ] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -42,6 +42,16 @@ Unreleased Changes
 Breaking Changes
 ================
 
+- Remap CrateDB :ref:`object_data_type` array data type from the PostgreSQL
+  JSON to JSON array type. That might effect some drivers that use the
+  PostgreSQL wire protocol to insert data into tables with object array typed
+  columns. For instance,  when using the ``Npgsql`` driver, it is not longer
+  possible to insert an array of objects into a column of the object array
+  data type by using the parameter of a SQL statement that has the JSON data
+  type and an array of CLR as its value. Instead, simply use a string array
```
Comment (mfussenegger) [inline docs/appendices/release-notes/unreleased.rst]:
```suggestion
  type and an array of CLR as its value. Instead, use a string array
```

**Comment by mfussenegger** [comment]
I'm wondering if we should allow casts from `string` to `object[]`, to mitigate the impact of this a bit?

**Comment by kovrus** [comment]
@mfussenegger allowing the `string` to `object[]` conversion in core would mean, for the Npgsql driver, adding yet another way of insertion an array of objects and would like something like that 
```
cmd.Parameters.AddWithValue("@obj_arr", NpgsqlTypes.NpgsqlDbType.Varchar, "[ { \"inner\": \"Zoon1\" } ]");
```
but then we would need to make `ArrayType` able to handle JSON array strings. Or you had something else on your mind?

To keep it as before, we would have to allow the `object` -> `object[]` conversion, then we can still do smth like that without any further adjustments in `ArrayType`:
```
cmd.Parameters.AddWithValue("@arr", NpgsqlTypes.NpgsqlDbType.Json, "[ { \"inner\": \"Zoon1\" } ]");
```

but i am not sure about the impact of adding this conversion.

**Comment by mfussenegger** [comment]
> To keep it as before, we would have to allow the object -> object[] conversion, then we can still do smth like that without any further adjustments in ArrayType:

Ah okay, I think I misunderstood how it worked before.

Let's leave it as it is then. Allowing `object -> object[]` doesn't make sense.

**Pull request merged**